src/openforms/contrib/haal_centraal/clients/__init__.py

```diff
@@ -21,5 +52,8 @@ def get_brp_client(**kwargs) -> BRPClient:
             f"No suitable client class configured for API version {version}"
         )
 
-    service_client_factory = ServiceClientFactory(service)
+    service_client_factory = BRPServiceClientFactory(
+        haal_centraal_config=config,
+        x_origin_oin_header=global_config.x_origin_oin_header,
+    )
     return ClientCls.configure_from(service_client_factory, **kwargs)
```

Comment:
Isn't it what the client factory is for? If we pass everything in `kwargs` so that it get picked up in the `__init__` then I feel we are loosing the purpose of the factory + `configure_from`. On the other side, having everything in the `__init__` makes it more explicit, as this factory pattern makes me think about Java 🤢

Reply:
I'll take another look after lunch - maybe it's the combination of this + the pre-request hook plugin that caused it to be overwhelming.

I'm thinking a little bit from the perspective of unit tests if we use the client class/instance directly or exclusively go through `get_brp_client`. If it's the former, making it an `__init__` parameter makes more sense from the perspective of loose coupling, if it's the latter, it's more integrated/self-contained. Let's discuss after lunch!